code/modules/mob/living/simple_animal/hostile/cockroach.dm

```diff
@@ -106,3 +106,107 @@
 		return TRUE
 	living_target.visible_message("<span class='notice'>[living_target] squashes [cockroach], not even noticing its spike.</span>", "<span class='notice'>You squashed [cockroach], not even noticing its spike.</span>")
 	return FALSE
+
+/mob/living/simple_animal/hostile/cockroach/glockroach/prime
+	name = "glockroach prime"
+	desc = "The ultimate form of the glockroach. Featuring altered mod polarities that offer greater customization."
+	icon_state = "glockroach-prime"
+	gold_core_spawnable = NO_SPAWN
+	health = 2 //extra armor
+	maxHealth = 2
+	ranged_cooldown_time = 20 //shuuts faster
```

Comment:
You said this wasn't working, and said something about delta time, could you elaborate further?

Reply:
Basically, ranged_cooldown_time does absolutely nothing. From what i could tell in the hostile mob code, it seems to just be world time being screwy as nothing else seems to be out of place.